modules/Kconfig.mcux

```diff
@@ -10,6 +10,12 @@ config HAS_MCUX
 
 if HAS_MCUX
 
+config MCUX_CORE_SUFFIX
+	string
+	help
+	  String describing the core identifer used by MCUX SDK when using
+	  dual core parts
```

Comment:
I posted this in the other PR but why is this needed, why can't your module check for `CPU_CORTEX_A53` (or similar)?

Reply:
The module could check for this- but it results in an (IMO) ugly `if` and `elseif` ladder in the module, because we don't always use the same suffix even if the core is the same.

IE on our dual core cm33 parts, we use suffixes like `_cm33_core0` and `_cm33_core1`, but on dual core parts that have an Xtensa core and CM33, the suffix would be `_cm33`. The end result of this is an `if`/`elseif` ladder like this:

```
if (CONFIG_CPU_LPC55S69_CPU0)

elseif(CONFIG_CPU_LPC55S69_CPU1)

elseif(CONFIG_CPU_MIMXRT595S_CM33)

....
```

Reply:
Naturally when combining two systems, like Zephyr's Kconfig and a HAL like NXP's, (which is probably designed externally to Zephyr) then some way of mapping between Zephyr and the HAL is needed.

And on the positive side it's good that this happens in the module's glue code.
But I take this new Kconfig and the old (mis-?)use of the CONFIG_SOC name as a sign that the existing implementation / mapping has not been done in a very clean way. (Ref: https://github.com/zephyrproject-rtos/hal_nxp/blob/d45b14c198d778658b7853b48378d2e132a6c4be/mcux/CMakeLists.txt#L15-L50)

Usually I wouldn't be fond of dealing with a Kconfig `SUFFIX` setting like this, but I understand the need for a mapping, and keeping this isolated to the glue layer makes it acceptable.

However, on the long term I think it would be nice to investigate if this could be made even cleaner.